ci/infra/azure/README.md

```diff
@@ -9,11 +9,25 @@ to deploy CaaS Platform on top of it.
 This document focuses on the key aspects of the infrastructure created
 by terraform.
 
-# Cluster layout
+# Cluster Setup
 
-## Setup service principal for terraform login credential
+## Service Principle
 
-Following the guide [Creating a Service Principal in the Azure Portal](https://www.terraform.io/docs/providers/azurerm/guides/service_principal_client_secret.html#creating-a-service-principal-in-the-azure-portal), and set up `ARM_CLIENT_ID`, `ARM_CLIENT_SECRET`, `ARM_SUBSCRIPTION_ID`, `and ARM_TENANT_ID` in `container-openrc.sh`.  Source `container-openrc.sh` before deploying terraform script.
+For cluster provision with terraform requires to have a service principle as authentication method.
+
+Follow the instruction to [Creating a Service Principal in the Azure Portal](https://www.terraform.io/docs/providers/azurerm/guides/service_principal_client_secret.html#creating-a-service-principal-in-the-azure-portal).
+
+You can also use Azure CLI to automate the task. For installation instructions, refer to link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install the Azure CLI].
+```bash
+az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/<SUBSCRIPTION_ID>"
+```
+
+Update and source `container-openrc.sh` before deploying terraform script.
+```bash
+source container-openrc.sh
+```
```

Comment:
Is this for Azure? it seems like for `openstack`?

Reply:
Yes this is for Azure. I believe the idea did came from openstack, right @cclhsu ?
ref: https://github.com/SUSE/skuba/blob/master/ci/infra/azure/container-openrc.sh

Reply:
Do we really need this? @cclhsu I think we already use the service principal to do the authentication instead of environment variables.

Reply:
There are two possible ways for terraform takes credential configs:
* environment variable (recommended) <- where we are
* in-line
ref: https://www.terraform.io/docs/providers/azurerm/guides/service_principal_client_secret.html

Reply:
Yes, I misread here :(, the thing is correct as what we decided to do. Thanks for clarifying. All good now thanks @c3y1huang and @cclhsu .

Reply:
Yes.  Same method is used for openstack and vsphere, and the question was also raised during demo (I think for vshpere) if this way is possible.

Reply:
Thanks for clarifying @cclhsu :+1: